control-plane/controllers/configentry_controller.go

```diff
@@ -381,6 +389,52 @@ func (r *ConfigEntryController) nonMatchingMigrationError(kubeEntry common.Confi
 	return fmt.Errorf("migration failed: Kubernetes resource does not match existing Consul config entry: consul=%s, kube=%s", consulJSON, kubeJSON)
 }
 
+// assignServiceVirtualIPs manually sends the ClusterIP for a matching service for ServiceRouter or ServiceSplitter
+// CRDs to Consul so that it can be added to the virtual IP table. The assignment is skipped if the matching service
+// does not exist or if an older version of Consul is being used. Endpoints Controller, on service registration, also
+// manually sends a ClusterIP when a service is created. This increases the chance of a real IP ending up in the
+// discovery chain.
+func assignServiceVirtualIP(ctx context.Context, logger logr.Logger, consulClient *capi.Client, crdCtrl Controller, namespacedName types.NamespacedName, configEntry common.ConfigEntryResource) error {
+
+	// Only assign the virtual IP for service routers and splitters. Every other type is skipped.
+	kubeKind := configEntry.KubeKind()
+	if kubeKind != common.ServiceRouter && kubeKind != common.ServiceSplitter {
+		return nil
+	}
+
+	service := corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      configEntry.KubernetesName(),
+			Namespace: namespacedName.Namespace,
+		},
+	}
+
+	err := crdCtrl.Get(ctx, namespacedName, &service)
```

Comment:
nit
```suggestion
	if err := crdCtrl.Get(ctx, namespacedName, &service); err != nil {
```
we try and use this pattern if the only param on the left is the error. We can do this for L424 as well

Reply:
Thank you. I knew there was a more idiomatic way but it was slipping my mind.